docs: assume core repositories are untapped (#15693)
Homebrew/brew: docs/Formula-Cookbook.md

Make sure you run `brew update` before you start. This ensures your Homebrew installation is a Git repository.

> To create or edit formulae locally, you'll need to `brew tap homebrew/core` if you haven't previously. This taps `homebrew/core`, creating a Git repository in `$(brew --repository homebrew/core)`. As you are developing, you'll also need to set `HOMEBREW_NO_INSTALL_FROM_API=1` before any `install`, `reinstall` or `upgrade` commands, to force `brew` to use the local repository instead of the API.

Reviewer: ```suggestion
To create or edit formulae locally, you'll need to `brew tap homebrew/core` if you haven't previously. This clones the `homebrew/core` Git repository to `$(brew --repository homebrew/core)`. As you are developing, you'll also need to set `HOMEBREW_NO_INSTALL_FROM_API=1` before any `install`, `reinstall` or `upgrade` commands, to force `brew` to use the local repository instead of the API.
```

Reviewer 2: I wonder if it's worth mentioning `tap --force` here too.

Reviewer 3: I’m not sure when `--force` might be needed. I wanted to note that, currently, the documentation mentions [`brew tap homebrew/core`](https://github.com/search?q=repo%3AHomebrew%2Fbrew%20%22brew%20tap%20homebrew%2Fcore%22&type=code) in two places (and in a script), and [`brew tap --force homebrew/core`](https://github.com/search?q=repo%3AHomebrew%2Fbrew+%22brew+tap+--force+homebrew%2Fcore%22&type=code) in one place (and in the Dockerfile).

Author: `--force` will eventually be needed if `HOMEBREW_NO_INSTALL_FROM_API=1` is not set.

Reviewer 3: I'm not sure what you mean by eventually. The current behavior does not require `--force` even if `HOMEBREW_NO_INSTALL_FROM_API` is not set. Here is a minimal example on the master branch:
```console
$ podman run --rm -it homebrew/ubuntu22.04:master
linuxbrew@8fbdd629d040:~$ brew tap
homebrew/core
linuxbrew@8fbdd629d040:~$ brew tap homebrew/cask
Running `brew update --auto-update`...
==> Homebrew collects anonymous analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics have been recorded yet (nor will be during this `brew` run).

==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

Warning: Tapping homebrew/cask is no longer typically necessary.
Add --force if you are sure you need one.
==> Tapping homebrew/cask
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 759853, done.
remote: Counting objects: 100% (11425/11425), done.
remote: Compressing objects: 100% (212/212), done.
remote: Total 759853 (delta 11293), reused 11334 (delta 11213), pack-reused 748428
Receiving objects: 100% (759853/759853), 340.86 MiB | 9.83 MiB/s, done.
Resolving deltas: 100% (541116/541116), done.
Tapped 4245 casks (4,319 files, 368.0MB).
linuxbrew@8fbdd629d040:~$ brew tap
homebrew/cask
homebrew/core
linuxbrew@8fbdd629d040:~$ env | grep HOMEBREW                    
linuxbrew@8fbdd629d040:~$ 
```

The same thing locally:
```console
$ brew untap homebrew/core
Error: No available tap homebrew/core.
$ brew tap homebrew/core 
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
==> Updated Homebrew from 5226cd434 to 0e8d1e0be.
Updated 2 taps (homebrew/core and homebrew/cask).
==> New Formulae
mvfst
==> New Casks
keyclu                                   lm-studio

You have 5 outdated formulae and 2 outdated casks installed.

Warning: Tapping homebrew/core is no longer typically necessary.
Add --force if you are sure you need one.
==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 1554395, done.
remote: Counting objects: 100% (125/125), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 1554395 (delta 73), reused 75 (delta 43), pack-reused 1554270
Receiving objects: 100% (1554395/1554395), 405.42 MiB | 8.72 MiB/s, done.
Resolving deltas: 100% (1091356/1091356), done.
Updating files: 100% (7139/7139), done.
Tapped 3 commands and 6781 formulae (7,151 files, 470.2MB).
$ env | grep HOMEBREW
HOMEBREW_PREFIX=/usr/local
HOMEBREW_CELLAR=/usr/local/Cellar
HOMEBREW_REPOSITORY=/usr/local/Homebrew
HOMEBREW_GITHUB_API_TOKEN=[REDACTED]
$ 
```

I've not personally encountered a situation where I needed `--force` to tap, before or after the install from API change, with or without `HOMEBREW_NO_INSTALL_FROM_API` being set.

Reviewer 4: In 4.2.0 the warning will become an error.

Reviewer 2: @EricFromCanada any thoughts on updating to use `brew tap --force`? I'm fine if you're 👎🏻, just wanted to check explicitly either way.
